x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/resources_stats_collector.ts

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { Logger } from '@kbn/core/server';
+import { SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import { LATEST_FINDINGS_INDEX_DEFAULT_NS } from '../../../../common/constants';
+import type { CspmResourcesStats } from './types';
+
+interface ResourcesStats {
+  accounts: {
+    buckets: AccountEntity[];
+  };
+}
+
+export interface AccountEntity {
+  key: string; // account_id
+  doc_count: number;
+  resource_type: {
+    buckets: ResourceType[];
+  };
+}
+
+interface ResourceType {
+  key: string; // resource_type
+  doc_count: number;
+  resource_sub_type: {
+    buckets: ResourceSubType[];
+  };
+}
+
+interface ResourceSubType {
+  key: string; // resource_sub_type
+  doc_count: number;
+  evaluation: {
+    buckets: EvaluationStats[];
+  };
+}
+
+interface EvaluationStats {
+  key: string; // passed / failed
+  doc_count: number;
+}
+
+export const getResourcesStats = async (
+  esClient: ElasticsearchClient,
+  logger: Logger
+): Promise<CspmResourcesStats[]> => {
+  try {
+    const isIndexExists = await esClient.indices.exists({
+      index: LATEST_FINDINGS_INDEX_DEFAULT_NS,
+    });
+
+    if (isIndexExists) {
+      return await getResourceStats(esClient, LATEST_FINDINGS_INDEX_DEFAULT_NS);
+    }
+
+    return [];
+  } catch (e) {
+    logger.error(`Failed to get resources stats ${e}`);
+    return [];
+  }
+};
+
+const getResourceStats = async (
+  esClient: ElasticsearchClient,
+  index: string
+): Promise<CspmResourcesStats[]> => {
+  const resourcesStatsResponse = await esClient.search<unknown, ResourcesStats>(
+    getResourcesStatsQuery(index)
+  );
+  const resourcesStats = resourcesStatsResponse.aggregations
+    ? parseResourcesStatsQueryResponse(resourcesStatsResponse.aggregations)
+    : [];
+  return resourcesStats;
+};
+
+const parseResourcesStatsQueryResponse = (resourcesStats: ResourcesStats): CspmResourcesStats[] => {
+  const accounts = resourcesStats.accounts.buckets;
+
+  const parsedResourcesStats = accounts.map((account) => {
```

Comment:
nit:
i might be wrong but parsing is usually done with words or sentences, strings in our case. here its just creating an array, i dont think that's parsing. consider rename

Reply:
since the return type ends up being `CspmResourcesStats`, the function could be renamed to `getCspmResourcesStats`

then we could also change line 75 and call the `resourcesStats` variable `cspmResourcesStats` so it's not the same as the param in `parseResourcesStatsQueryResponse` which makes input/output names the same.